File: twitter-crawl/model/twitter-crawl.go
package model

import (
	"time"
)

type TwitterCrawl struct {
	CommonEntity

	ID        uint      `gorm:"column:id;primaryKey;autoIncrement:true" json:"id"`
	TweetID   string    `gorm:"column:tweet_id;type:varchar(256);not null;unique" json:"tweet_id"`
	UserID    string    `gorm:"column:user_id;type:varchar(256);not null" json:"user_id"`
	Username  string    `gorm:"column:username;type:varchar(256);not null" json:"username"`
	TweetTime time.Time `gorm:"column:tweet_time;precision:6" json:"tweet_time"`
	Content   string    `gorm:"column:content;type:text;not null" json:"content"`
}

func (TwitterCrawl) TableName() string {
	return "twitter_crawl"
}

File: backend/src/model/twitter-crawl.ts
import { Column, Entity, Index, PrimaryGeneratedColumn } from 'typeorm'
import { CommonEntity } from './common'

@Entity()
export class TwitterCrawl extends CommonEntity {
  @PrimaryGeneratedColumn()
  id: number

  @Column('varchar', { length: 256, unique: true })
  tweet_id: number

  @Column('varchar', { length: 256 })
  user_id: number

  @Column('varchar', { length: 256 })
  username: string

  @Column('timestamp', { default: null })
  timestamp: Date

  @Column('text')
  content: string
}
